main/src/com/google/refine/expr/functions/Cross.java

```diff
@@ -71,22 +84,27 @@ public Object call(Properties bindings, Object[] args) {
 
                 if (v instanceof WrappedCell) {
                     return lookup.getRows(((WrappedCell) v).cell.value);
+                } else if (v instanceof Number && targetColumnName.equals(INDEX_COLUMN_NAME)){
+                    return lookup.getRows(((Number) v).intValue());
                 } else {
                     return lookup.getRows(v);
                 }
             }
         }
-        return new EvalError(ControlFunctionRegistry.getFunctionName(this) + " expects a cell or cell value, a project name to look up, and a column name in that project");
+        return new EvalError(ControlFunctionRegistry.getFunctionName(this) + " expects a cell or cell value, a project name to look up (optional), and a column name in that project (optional)");
     }
```

Comment:
I would change `cell value` for just value everywhere in the doc. It's just a value, it can come from anywhere (even literals values are accepted!).